tests/test_optional.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any
```

Comment:
I'll let @omry comment on this, but I'm not sure it's worth adding a new test file just for this (could be part of `test_basic_ops_{list,dict}`)
(don't change it just based on my opinion though)

Reply:
My motivation was to be able to write `Optional[...]` tests in one place instead of having to split them up according to `Dict[...,Optional]` / `List[Optional]` / `Optional[User]` / ...
I'm hoping to make more improvements to optional type handling in the near future.

Reply:
(Maybe something like `test_typing.py` would be more future-proof...)

Reply:
That's fine by me => will just let @omry resolve this point.

Reply:
Resolved in https://github.com/omry/omegaconf/pull/749/commits/7f5892e45897c65ca2b8fcda876e83b792dffd67: I've put the new tests in `test_base_config.py`.